src/scalar_metrics.go

```diff
@@ -56,6 +41,24 @@ func populateScalarMetrics(device string, metricSet metricSet, entity *integrati
 		log.Error(err.Error())
 	}
 
+	snmpGetResult, err := theSNMP.Get(oids)
+	if err != nil {
+		return err
+	}
+
+	// Response received with errors
+	if snmpGetResult.Error != gosnmp.NoError {
+		err = ms.SetMetric("errorCode", getErrorCode(snmpGetResult.Error), metric.ATTRIBUTE)
+		if err != nil {
+			log.Error(err.Error())
+		}
+		err = ms.SetMetric("errorMessage", getErrorMessage(snmpGetResult.Error), metric.ATTRIBUTE)
+		if err != nil {
+			log.Error(err.Error())
+		}
+		return nil
```

Comment:
Are you sure about returning nil? We are actually having an error from the response and not continuing the execution. What about returning a new error here?